src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/utilities/SystemDictionaryHelper.java

```diff
@@ -91,9 +89,7 @@ public int compare(InstanceKlass k1, InstanceKlass k2) {
          }
       }
 
-      Object[] tmpArray = tmp.toArray();
-      InstanceKlass[] searchResult = new InstanceKlass[tmpArray.length];
-      System.arraycopy(tmpArray, 0, searchResult, 0, tmpArray.length);
+      InstanceKlass[] searchResult = tmp.toArray(new InstanceKlass[0]);
```

Comment:
Is it too far outside the scope of these changes to make tmp an `ArrayList` rather than a `Vector`?

Reply:
I'll create separate PRs to migrate Vector usages to ArrayList.